python/xir/parser.py

```diff
@@ -212,7 +214,7 @@ def gatestmt(self, args):
             wires = args[1][1]
 
         stmt_options = {
-            "ctrl_wires": ctrl_wires,
+            "ctrl_wires": tuple(ctrl_wires),
```

Comment:
What do you think about sorting these wires as a nice convenience?
```suggestion
            "ctrl_wires": tuple(sorted(ctrl_wires)),
```

Reply:
Same comment as above regarding keeping the order. Otherwise I don't mind sorting them. I don't think this would work though if there's a mix of strings and integers, right?

Reply:
Not sure if this is a bit overkill, but we could sort it like this:
```python
sorted(ctrl_wires, key=lambda x: sum([10**ord(i) for i in str(x)]))
```
which should work for integers and string (not for floats though if we decide we want to include that).

Reply:
> I don't think this would work though if there's a mix of strings and integers, right?

Oh, right, good point!

> Not sure if this is a bit overkill, but we could sort it like this:
> 
> ```python
> sorted(ctrl_wires, key=lambda x: sum([10**ord(i) for i in str(x)]))
> ```
> 
> which should work for integers and string (not for floats though if we decide we want to include that).

Yeah, that works, but I think we can agree this is a bit hacky.  I should clarify that my primary motivation for sorting the wires is deterministic serialization (when `ctrl_wires` is a set).  With this in mind, we can just use:

```python
sorted(ctrl_wires, key=hash)
```

Reply:
Oh, yes, in that case I'd say the hash would be a good way to sort it by. I was mainly thinking it'd be nice to always return everything sorted e.g. by ints and then strings alphabetically.